packages/dev/core/src/Engines/WebGPU/webgpuShaderProcessorsGLSL.ts

```diff
@@ -331,7 +331,7 @@ export class WebGPUShaderProcessorGLSL extends WebGPUShaderProcessor {
             const lastClosingCurly = code.lastIndexOf("}");
             code = code.substring(0, lastClosingCurly);
             code += "gl_Position.y *= yFactor_;\n";
-            if (!engine.isNDCHalfZRange) {
+            if (engine && !engine.isNDCHalfZRange) {

```

Comment:
I guess it's a valid use case not to have engine defined (else we would not test for it)? In this case, shouldn't `isNDCHalfZRange` be retrieved from elsewhere? Else, it means that when engine is not defined, it's not possible to setup the shader processor for WebGPU usage.

Reply:
Doing the work for WebGPU was a bit challenging because of the needed dependencies. I forgot to mention in the docs, but this PR is currently WebGL only. What I can do is create a small state-object for the webgpu engine (just like i did for the thinengine) to keep variables like this.